### PR TITLE
SCRIPTS: Say It with Flowers

### PR DESCRIPTION
**Pull request by waytim**
http://ffxiclopedia.wikia.com/wiki/Say_It_with_Flowers

In response to https://github.com/DarkstarProject/darkstar/issues/2012

Tested and follows wiki.  
Does take priority over other quests as stated in wiki.

### CODE REVIEW

File: scripts/zones/Windurst_Waters/npcs/Kenapa-Keppa.lua [hunk 1]
```diff
@@ -20,25 +20,25 @@ require("scripts/zones/Windurst_Waters/TextIDs");
...
-				player:startEvent(0x014b);
-			else
-				player:startEvent(0x014a,120);
-			end
-		elseif (count == 1 and trade:hasItemQty(4409,1) == true and gil == 0 and KenapaFood == 3) then	-- Traded item after receiving order
-			player:startEvent(0x0147,120);
-		end
-	end
+    FoodForThought = player:getQuestStatus(WINDURST,FOOD_FOR_THOUGHT);
+    KenapaFood = player:getVar("Kenapa_Food_var"); -- Variable to track progress of Kenapa-Keppa in Food for Thought
```
Comment (xdemolish) [inline scripts/zones/Windurst_Waters/npcs/Kenapa-Keppa.lua]: i know this wasn't part of your change, but make FoodForThought and KenapaFood local variables (local as in lua local, not dsp localvar)

File: scripts/zones/Windurst_Waters/npcs/Kenapa-Keppa.lua [hunk 2]
```diff
@@ -20,25 +20,25 @@ require("scripts/zones/Windurst_Waters/TextIDs");
...
-				player:startEvent(0x014b);
-			else
-				player:startEvent(0x014a,120);
-			end
-		elseif (count == 1 and trade:hasItemQty(4409,1) == true and gil == 0 and KenapaFood == 3) then	-- Traded item after receiving order
-			player:startEvent(0x0147,120);
-		end
-	end
+    local FoodForThought = player:getQuestStatus(WINDURST,FOOD_FOR_THOUGHT);
+    local KenapaFood = player:getVar("Kenapa_Food_var"); -- Variable to track progress of Kenapa-Keppa in Food for Thought
```
Comment (waytim) [inline scripts/zones/Windurst_Waters/npcs/Kenapa-Keppa.lua]: Changed to local vars.

File: scripts/zones/Windurst_Waters/npcs/Kenapa-Keppa.lua [hunk 3]
```diff
@@ -47,100 +47,103 @@ end;
...
+    SayFlowers = player:getQuestStatus(WINDURST,SAY_IT_WITH_FLOWERS);
+    FlowerProgress = player:getVar("FLOWER_PROGRESS"); -- Variable to track progress of Say It with Flowers.
+    hatstatus = player:getQuestStatus(WINDURST,HAT_IN_HAND);
+    KenapaFood = player:getVar("Kenapa_Food_var"); -- Variable to track progress of Kenapa-Keppa in Food for Thought
+    KenapaOvernight = player:getVar("Kenapa_Overnight_var"); -- Variable to track progress for Overnight Delivery
+    KenapaOvernightDay = player:getVar("Kenapa_Overnight_Day_var"); -- Variable to track the day the quest is started.
+    KenapaOvernightHour = player:getVar("Kenapa_Overnight_Hour_var"); -- Variable to track the hour the quest is started.
+    needToZone = player:needToZone();
+    pFame = player:getFameLevel(WINDURST);
+    HourOfTheDay = VanadielHour();
```
Comment (xdemolish) [inline scripts/zones/Windurst_Waters/npcs/Kenapa-Keppa.lua]: o god how did i not see these
(they need local'ing too)

**Comment by waytim** [comment]
Updated; made sure everything is localed out.

**Pull request merged**